app/models/service_template_transformation_plan_task.rb

```diff
@@ -81,6 +82,20 @@ def preflight_check
     { :status => 'Error', :message => error.message }
   end
 
+  def preflight_check_vm_exists_in_destination
+    send("preflight_check_vm_exists_in_destination_#{destination_ems.emstype}")
+  end
+
+  def preflight_check_vm_exists_in_destination_rhevm
+    vms = Vm.where(:name => source.name, :ems_cluster => destination_cluster)
```

Comment:
Do you have the `destination_ems` here?

It would be better to say `destination_ems.vms_and_templates.where(:name => source.name, :ems_cluster => destination_cluster).count.zero?`

Also better to do `count.zero?` instead of actually bring back the vm objects just to later check if it is empty

Reply:
Thanks. I updated the code for both RHV and OpenStack providers.